### DISCUSSION TITLE
Equivalent for RfcSessionManager BeginContext and EndContext

### DISCUSSION THREAD

**Discussion by garuda64** [dbosoft/YaNco#274] [Q&A] [answered]
Hello, thank you very much for supporting this very useful library, I have a question:

What is the equivalent to execute this function:

```
IRfcFunction commit = repo.CreateFunction("BAPI_TRANSACTION_COMMIT");
var pt = repo.CreateFunction("ZSD_RFC_SGR_PEDIDOS_PAC");
var Items = pt.GetTable("POSICIONES");

pt.SetValue("USER_AUTH","Test");

RfcSessionManager.BeginContext(destination);

   pt.Invoke(destination);
   commit.Invoke(destination);

RfcSessionManager.EndContext(destination);
```

Thank you very much for the support and help.

**Comment by fw2568** [comment, marked as answer]
Hello,

Thanks for your kind words.

YaNco relies on the SAP Netweaver RFC library, which is stateful by default, so there is no need to enter a session. 

Quoted from the NW RFC 7.50 Programming Guide:

> We end this chapter by noting one fundamental difference the NW RFC library has compared
> with the other two connectors, SAP Java Connector (JCo 3.1) and SAP .NET Connector (NCo
> 3.0): every RFC client connection you open with the NW RFC library is stateful by default. In
> the other two connectors they are stateless by default and you have to put some effort into it,
> if you need a stateful connection. 

So, if you execute your functions in the same RFCContext, they will be in the same session. 

Best Regards,
Frank
